framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -853,9 +853,9 @@ private S upperBound(S other, boolean shouldWiden) {
             // other are discarded, as one of store implicitly contains 'top'
             // for that variable.
             FlowExpressions.LocalVariable localVar = e.getKey();
-            if (localVariableValues.containsKey(localVar)) {
+            V thisVal = localVariableValues.get(localVar);
```

Comment:
@smillst I gave all of these another look again and it looks like all these maps might have null values if the `Store` is a `LockStore`. Inside `CFAbstractStore`, we always make sure that only non-null new entry is added to these maps. However, for `LockStore`, we simply use the return result of `changeLockAnnoToTop` (internally calls `createAbstractValue` which can return `null`) without checking for nullness.

Reply:
The documentation for LockStore#changeLockAnnoToTop  makes it sound that it should also never return null. So I think the right thing to do is to see what goes wrong if you add a null check there, too.